### ISSUE TITLE
Add reranker support

### ISSUE THREAD

**Issue by LD-Reborn** [LD-Reborn/embeddingsearch#108]
**Is your feature request related to a problem? Please describe.**
idk it would be cool to do.

**Describe the solution you'd like**
After doing the initial retrieval and filtering to TopN, users should have the ability to have the results be re-ranked using a re-ranker.

**Describe alternatives you've considered**
none

**Additional context**
I'm thinking of the re-ranking being a separate stage after producing the result list.
Multiple re-rankers possible? Maybe. Would be cool. Experimentation needed.

**Comment by LD-Reborn** [comment]
Due to current hardware limitations on my side (i.e. not enough VRAM to test larger models like Qwen3-Reranker-8B on GPU with my larger test dataset) and results that were worse than (or at best matched) their non-reranked counterparts I have decided to put this issue to the side until I have better hardware.

The current state (i.e. a proof-of-concept-style setup) is committed in the branch 108-add-reranker-support if someone wants to take a closer look. (However it does not represent the approach I would like for a proper implementation)

Some thoughts on the planned (proper) implementation:
1. Due to reranker models requiring plain text instead of embeddings, the embeddings tree (searchdomain->entity->datapoint->embedding) cannot be used for this purpose. I.e. the to-be-reranked texts must be stored in the entity attributes (which finally also gives the attributes a functional purpose).
2. Due to the batching that is possible (and should be preferred) I would disregard the possibility of giving entities reranking options like probmethod etc. - Rather to keep things simple it should be a searchdomain-only set of settings that controls the reranking process.
3. Utilizing multiple rerankers for a single query (like it is already done with embeddings to leverage orthogonality) looks possible and must be explored.
4. Rerankers do not output results in $[0;1]$ range but in logits, i.e. ${\displaystyle (-\infty ,+\infty )}$ which needs to be remapped (e.g. `1 / (1 + math.exp(-x))`)
